encoder.go

```diff
@@ -32,8 +32,20 @@ func NewGenericStreamEncoder(w io.Writer, marshalFn Marshaler, array bool) *Stre
 	}
 }
 
+func (sw *StreamEncoder) SetFirst(first bool) *StreamEncoder {
+	sw.first = first
+	return sw
+}
+
+func (sw *StreamEncoder) SetArray(array bool) *StreamEncoder {
+	sw.array = array
+	return sw
+}
+
 var commaNl = []byte(",\n")
 
+const bufLimit = 64 * 1024 // 64kB. Need more testing on this number.
```

Comment:
I'd make this a variable on the StreamEncoder struct and let this be the default.

Reply:
Okay